### DISCUSSION TITLE
"Stub HTTP requests"

### DISCUSSION THREAD

**Discussion by glaucocustodio** [evenbetterspecs/evenbetterspecs.github.io#18] [General]
What are your thoughts on https://evenbetterspecs.github.io/#stub-http-requests?

**Comment by v-yarotsky** [comment]
I prefer to wrap interactions with HTTP services in a Client class, and test that with VCR. The rest of the system is tested against a stub Client. This is both easier for me to use, and helps ensure that all HTTP interactions with an external service happen in one place, making it easy to add logging and instrumentation. I've seen some codebases with http requests sprinkled gratuitously throughout code, and test cases full of http stubs - would not recommend!

**Reply by glaucocustodio** [reply]
I agree with you, that's what I meant with this guideline, perhaps I should rephrase it to make clear my intention.. Do you want to make a PR for that @v-yarotsky?
